allensdk/test/model/test_biophysical_all_active.py

```diff
@@ -0,0 +1,142 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+import pytest
+import os
+import numpy
+from allensdk.model.biophys_sim.config import Config
+from allensdk.model.biophysical.utils import Utils, AllActiveUtils
+from allensdk.api.queries.biophysical_api import BiophysicalApi
+from allensdk.core.dat_utilities import DatUtilities
+from allensdk.ephys import ephys_features
+
+@pytest.mark.xfail(raises=ImportError,reason='neuron installation required')
+def test_biophysical_aa():
+    """
+    Test for backward compatibility of the legacy all-active models
+    """
+    neuronal_model_id = 497233125    # get this from the web site
+
+    model_directory = 'aa_model'
+
+    bp = BiophysicalApi('http://api.brain-map.org')
+    bp.cache_stimulus = False  # don't want to download the large stimulus NWB file
+    bp.cache_data(neuronal_model_id, working_directory=model_directory)
+    os.chdir(model_directory)
+    os.system('nrnivmodl modfiles')
+
+    description = Config().load('manifest.json')
+    utils = AllActiveUtils(description)
+    h = utils.h
+
+    manifest = description.manifest
+    morphology_path = manifest.get_path('MORPHOLOGY')
+    utils.generate_morphology(morphology_path.encode('ascii', 'ignore').decode("utf-8"))
+    utils.load_cell_parameters()
+
+    stim = h.IClamp(h.soma[0](0.5))
+    stim.amp = 0.33   # Sweep 46
+    stim.delay = 1000.0
+    stim.dur = 1000.0
+
+    h.tstop = 3000.0
+
+    vec = utils.record_values()
+
+    h.finitialize()
+    h.run()
+
+    junction_potential = description.data['fitting'][0]['junction_potential']
+    ms = 1.0e-3
+
+    output_data = (numpy.array(vec['v']) - junction_potential) # in mV
+    output_times = numpy.array(vec['t']) * ms # in s
+    output_path = 'output_voltage.dat'
+
+    DatUtilities.save_voltage(output_path, output_data, output_times)
+    
+    num_spikes = len(ephys_features.detect_putative_spikes(output_data,output_times))
+    assert num_spikes == 18 # taken from the web app where the legacy model output is shown
+    os.chdir(os.pardir)
+
+
+@pytest.mark.xfail(raises=ImportError,reason='neuron installation required')
+def test_biophysical_peri():
+    """
+    Test for backward compatibility of the perisomatic models
+    """
+    neuronal_model_id = 482934212    # get this from the web site
+
+    model_directory = 'peri_model'
+
+    bp = BiophysicalApi('http://api.brain-map.org')
```

Comment:
Calling to the warehouse like this is pretty bad news in a test. This is because:
1. it can fail randomly if e.g the warehouse goes down or is slow to respond
2. the test is likely to be slow, since it is waiting on an http request

If the test data you are getting are not large, I suggest just committing them to a subdirectory here. If they are large, let's try to think up a solution.

Reply:
Not too big. I will create a subdirectory for the test.